README.markdown

```diff
@@ -5010,6 +5011,17 @@ This function was first introduced in the `v0.5.0rc1` release.
 
 [Back to TOC](#nginx-api-for-lua)
 
+ngx.req.set\_cong
```

Comment:
I think we should make it a method of the cosocket object so that we can set socket-level congestion control algorithms for the upstream cosockets as well. This would be much more useful and general.

Reply:
@agentzh is there some document about cosocket:setoption and cosocket? Or I should read the code?

Reply:
@blacktear23 See https://github.com/openresty/lua-nginx-module#tcpsocksetoption It's in the official documentation.

Reply:
@agentzh After I do some research on let tcpsock:setoption can set congestion control I found I cannot change request's congestion control by below:
```nginx
access_by_lua '
    local sock, err = ngx.req.socket()
    if sock then
        sock.setoption("congestion", "reno")
    else
        if err then
            ngx.log(ngx.ERR, "error: ", err)
        end
    end
'
```

And error.log will found something like `content_by_lua(nginx.conf:42):9: error: no body,`

Reply:
@blacktear23 I think the error message is self-explanatory. You need a proper request body to use normal req cosockets. Otherwise, you may want to use "raw" req cosockets instead. Please read the official documentation for more details.

Reply:
@agentzh I do read the document and I found if I set raw to true, I will get error message as below: `content_by_lua(nginx.conf:42):6: attempt to call field 'setoption' (a nil value)`

Reply:
I found the problem.